packages/core/src/browser/window/default-window-service.ts

```diff
@@ -61,6 +67,21 @@ export class DefaultWindowService implements WindowService, FrontendApplicationC
         return confirmExit !== 'always';
     }
 
+    protected registerUnloadListener(): void {
+        window.addEventListener('unload', this.fireUnload.bind(this));
```

Comment:
Is there any advantage to firing `onUnload` on the `unload` event? My reasoning is that when `this.canUnload()` returns `true`, it will be fired in `beforeunload`. On electron, it will also be fired on `beforeunload` when `this.canUnload()` returns `false`. Only on the browser when `this.canUnload()` returns `false` will the event be fired on `unload`.

See my other comment for my proposed change.